include/rocksdb/file_system.h

```diff
@@ -119,7 +133,9 @@ struct IOOptions {
         prio(IOPriority::kIOLow),
         rate_limiter_priority(Env::IO_TOTAL),
         type(IOType::kUnknown),
-        force_dir_fsync(force_dir_fsync_) {}
+        force_dir_fsync(force_dir_fsync_) {
+    SetProperty("list_files_only", "false");
```

Comment:
Umm, I think this is going to be a non-trivial, unnecessary CPU consumer. Usually the threshold for negligible is .01%, and we already spend well over that in IOOptions::*, mostly in the destructor. And this would give it more work to do. (Internal detail in P533216605)